src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -478,6 +487,11 @@ private static IEnumerable<string> EnumerateFilesRecursively(string path)
                         }
                     }
                 }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(dirBuffer);
+                    Interop.Sys.CloseDir(dirHandle);
```

Comment:
I know `CloseDir` will likely handle null fine, but I would check for it upfront too - out of the habit.